### DISCUSSION TITLE
bug(BarcodeReader): 条码扫描组件第一次打开加载事件中只有一个摄像设备，切换页面后才会加载出所有设备

### DISCUSSION THREAD

**Discussion by Linlccc** [dotnetcore/BootstrapBlazor#3872] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

有两个页面，两个页面都有使用BarcodeReader组件，第一次加载只有一个摄像设备，切换页面（不是刷新url的方式）后才会加载出所有设备

偶发性问题：
有时会高频触发结果事件，捂住摄像头也会触发，关闭摄像头后就不会触发了，再打开又会触发
参数一直都是相同的数字
该问题测试时无法100%重现

### Expected Behavior

第一次加载就加载出所有设备

### .NET Version

8.0.302

### Anything else?

运行系统：iphone
浏览器：safari

**Comment by bb-auto** [comment]
@Linlccc Thank you for filing this issue. In order for us to investigate this issue, please provide [a minimalistic repro project](https://github.com/dotnetcore/bootstrapblazor/blob/main/docs/repro.md) that illustrates the problem.

**Comment by ArgoZhang** [comment]
@densen2014 

我们需要把 https://www.blazor.zone/barcode-reader 网页内的功能与你的那个网站微信扫码那个功能一致即
打开后摄像头是全的，目前官网的例子只出现一个

**Reply by densen2014** [reply]
确认问题在于zxing代码,不适应新版ios. 需要点时间处理

**Comment by densen2014** [comment]
camera组件已经弄了，barcode reader估计没有，我一直以为说的是camera

**Reply by ArgoZhang** [reply]
@Linlccc @densen2014 我个人觉得出一个摄像头是没问题的

https://zxing-js.github.io/library/examples/multi-camera/

看这个最新版本的例子，`zxing` 内部有处理，把不相干或者不能用的已经屏蔽掉了

我们目前需要做的是应该更新 zxing 版本到最新

**Reply by densen2014** [reply]
然而我手机访问这个页面只出现一个设备。这是不对的
